model/delivery/form/LtiWizardForm.php

```diff
@@ -65,15 +64,9 @@ public function initElements()
         $classUriElt->setValue($class->getUri());
         $this->form->addElement($classUriElt);
 
-        /** @var ComplexSearchService $search */
-        $search = $this->getServiceManager()->get(ComplexSearchService::SERVICE_ID);
-        $queryBuilder = $search->query();
-        $query = $search->searchType($queryBuilder , ProviderService::CLASS_URI, true);
-        $queryBuilder->setCriteria($query);
-
-        $count = $search->getGateway()->count($queryBuilder);
-
-        if (0 === $count) {
+        /** @var LtiProviderService $ltiProviderService */
+        $ltiProviderService = $this->getServiceManager()->get(LtiProviderService::SERVICE_ID);
+        if ($ltiProviderService->count() === 0) {
```

Comment:
I think you can use `empty()` here

Reply:
That would mean something like `if (empty($ltiProviderService->findAll())) {`. I'm not sure about the improvement, since it would actually load all providers, which would imply for Rdf implementation 1 query to retrieve the Rdf resources + 1 query for each found resource to get the additional properties, and finally just to count them. I'd stick to the actual implementation with `->count()` which make only one query with a `SELECT COUNT().`
What do you think?

Reply:
I didn't check out the code so in this context maybe you are right. But I would not expect a new query to be fired every time I count the providers, it seems risky for me

Reply:
So would you cache it? What if you add a new provider and reload the page?

Reply:
Class property cache. It's caching only for the lifetime of the current request

Reply:
Ok then. Let's do that! Anyway, even if cached, I'd keep the `->count() === 0` call instead of `empty(->findAll())`, which is only one query.